commit ba1702cead159228cd3af8b92ed70a6b99d09611 (refs/changes/37/337/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-08-23T14:59:58-07:00 (3 years, 2 months ago)
    
    IMGAPI-582 imgapi-cli should have a --insecure option for an IMGAPI with a self-signed cert

diff --git a/bin/imgapi-cli b/bin/imgapi-cli
index 4d7cf58..f084ce7 100755
--- a/bin/imgapi-cli
+++ b/bin/imgapi-cli
@@ -55,7 +55,8 @@ function main(argv) {
         envopts: [
             ['IMGAPI_CLI_IDENTITY', 'identity'],
             ['IMGAPI_CLI_USER', 'user'],
-            ['IMGAPI_CLI_CHANNEL', 'channel']
+            ['IMGAPI_CLI_CHANNEL', 'channel'],
+            ['IMGAPI_CLI_INSECURE', 'insecure']
         ],
         connectTimeout: 10000
     });
diff --git a/bin/joyent-imgadm b/bin/joyent-imgadm
index 92061d9..7dee488 100755
--- a/bin/joyent-imgadm
+++ b/bin/joyent-imgadm
@@ -66,7 +66,8 @@ function main(argv) {
         features: FEATURES,
         envopts: [
             ['JOYENT_IMGADM_IDENTITY', 'identity'],
-            ['JOYENT_IMGADM_USER', 'user']
+            ['JOYENT_IMGADM_USER', 'user'],
+            ['JOYENT_IMGADM_INSECURE', 'insecure']
         ],
         connectTimeout: 10000
     });
diff --git a/bin/updates-imgadm b/bin/updates-imgadm
index cebf3d3..ff21077 100755
--- a/bin/updates-imgadm
+++ b/bin/updates-imgadm
@@ -71,7 +71,8 @@ function main(argv) {
         envopts: [
             ['UPDATES_IMGADM_IDENTITY', 'identity'],
             ['UPDATES_IMGADM_USER', 'user'],
-            ['UPDATES_IMGADM_CHANNEL', 'channel']
+            ['UPDATES_IMGADM_CHANNEL', 'channel'],
+            ['UPDATES_IMGADM_INSECURE', 'insecure']
         ],
         connectTimeout: 10000
     });
diff --git a/lib/cli.js b/lib/cli.js
index 4bf3dd8..85b9388 100644
--- a/lib/cli.js
+++ b/lib/cli.js
@@ -306,6 +306,8 @@ CLI.prototype.main = function main(argv, callback) {
         if (self._connectTimeout) {
             imgapiOpts.connectTimeout = self._connectTimeout;
         }
+        imgapiOpts.rejectUnauthorized = (opts.insecure === undefined
+            ? true : !opts.insecure);
         self.client = imgapi.createClient(imgapiOpts);
 
         var subcmd = args.shift();
@@ -334,7 +336,8 @@ CLI.prototype.handleArgv = function handleArgv(argv, envopts, callback) {
     var longOpts = this.longOpts = {
         'help': Boolean,
         'version': Boolean,
-        'debug': Boolean
+        'debug': Boolean,
+        'insecure': Boolean
     };
     var shortOpts = this.shortOpts = {
         'h': ['--help'],
@@ -428,7 +431,8 @@ CLI.prototype.printHelp = function printHelp(callback) {
         'Options:',
         '    -h, --help          Show this help message and exit.',
         '    --version           Show version and exit.',
-        '    -d, --debug         Verbose logging.'
+        '    -d, --debug         Verbose logging.',
+        '    --insecure          Do not validate a TLS certificate.'
     ]);
     if (this.auth === 'basic') {
         lines = lines.concat([
